Lib/bdb.py

```diff
@@ -26,6 +34,12 @@ def __init__(self, skip=None):
         self.frame_returning = None
 
     def canonic(self, filename):
+        """Get a filename into a canonical form.
```

Comment:
This function is not really a method, as it does not use 'self'.  But is was written before 'staticmethod' and the original author(s) must have not wanted to expose it at module level.  We are here documenting the API and will next test the current implementation.  Once tested, we could possibly revise.

1. `canonic` does 2 things different from os function(s).
1A. Leave angle bracketed names alone.  This following part of the doc, copied in your patch, "stripped of surrounding angle brackets" does not match the code.  Either way, I don't know what bracketed '<filename>'s are about or when, if ever, they occur.  I also don't know if the existing test is the fastest possible.
1B. Cache the mapping of non-bracketed names to absolute name.  This must be for speed as the same abbreviated name is used repeatedly.

2. It does not cache bracketed name.  If it did, I would say 'test alternatives for speed'.
3. `normcase` is not `normpath` and latter, called by `abspath`, does not call `normcase`.